### PR TITLE
MBA-29: Implement Login Feature

### PR DESCRIPTION
**Pull request by abid0-07**
Description:
This pull request introduces the Login Feature as part of the Event Management System. The feature allows registered users to securely log in to their accounts using their email and password. Upon successful authentication, users will be granted access to the system's features.

**Key Changes**:

1. Added a new `LoginForm` component for user authentication.
2. Implemented backend API endpoints for user login and token generation (JWT).
3. Integrated validation for required fields and email format.
4. Added password verification against hashed passwords stored in the database.
5. Implemented error handling for invalid credentials and other edge cases.
6. Added unit tests for the login functionality.

[Jira Link](https://diu-team-shamim366.atlassian.net/browse/MBA-29)

### CODE REVIEW

File: src/app/auth/login/page.tsx [hunk 1]
```diff
@@ -1,146 +1,195 @@
...
+  const handleMfaVerification = async (e: React.FormEvent) => {
+    e.preventDefault()
+    await verifyMfaCode(mfaCode)
+  }
 
-    if (!password.trim()) {
-      setErrors((prev) => ({ ...prev, password: "Password is required" }));
-      isValid = false;
-    }
+  const handleSocialLogin = async (provider: "google" | "facebook" | "microsoft") => {
```
Comment (shamim36) [inline src/app/auth/login/page.tsx]: We do not need any third party login platform. Remove it

File: src/app/auth/login/page.tsx [hunk 2]
```diff
@@ -1,146 +1,195 @@
...
+                  <Label htmlFor="mfa-code">Verification Code</Label>
+                  <Input
+                    id="mfa-code"
+                    placeholder="Enter 6-digit code"
+                    value={mfaCode}
+                    onChange={(e) => setMfaCode(e.target.value)}
+                    required
+                  />
+                </div>
+              </CardContent>
```
Comment (shamim36) [inline src/app/auth/login/page.tsx:82]: Our Backend do not have any two step verification method. Remove it

File: src/app/auth/login/page.tsx [hunk 3]
```diff
@@ -1,146 +1,195 @@
...
+              <Card>
+                <CardContent className="pt-6 space-y-4">
+                  <Button
+                    variant="outline"
+                    className="w-full justify-start"
+                    onClick={() => handleSocialLogin("google")}
+                    disabled={isLoading}
+                  >
+                    <SocialIcons.Google className="mr-2 h-4 w-4" />
+                    Continue with Google
```
Comment (shamim36) [inline src/app/auth/login/page.tsx]: Unnecessary code snipts

**Review by shamim36 — COMMENTED** [review]
There are few issues inside your code. Fix those issues
